SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery reproducibly fails in 7.x

**Build scan**:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-1/1560/

**Repro line**:
```
./gradlew ':server:test' --tests "org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery" -Dtests.seed=92425189D86F7872 -Dtests.security.manager=true -Dtests.locale=uk -Dtests.timezone=America/St_Thomas -Druntime.java=8
```

**Reproduces locally?**:
yes

**Applicable branches**:
7.x

**Failure history**:
Couldn't find any other recent failures

**Failure excerpt**:
```
ite: Test class org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests
  1> [2020-06-16T16:17:22,534][INFO ][o.e.i.q.SpanMultiTermQueryBuilderTests] [testToQueryInnerTermQuery] before test
  1> [2020-06-16T16:17:22,589][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] no modules loaded
  1> [2020-06-16T16:17:22,589][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] loaded plugin [org.elasticsearch.test.TestGeoShapeFieldMapperPlugin]
  1> [2020-06-16T16:17:23,605][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] no modules loaded
  1> [2020-06-16T16:17:23,606][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] loaded plugin [org.elasticsearch.test.TestGeoShapeFieldMapperPlugin]
  1> [2020-06-16T16:17:23,625][INFO ][o.e.i.q.SpanMultiTermQueryBuilderTests] [testToQueryInnerTermQuery] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery" -Dtests.seed=92425189D86F7872 -Dtests.security.manager=true -Dtests.locale=uk -Dtests.timezone=America/St_Thomas -Druntime.java=8
  2> java.lang.AssertionError: 
    Expected: an instance of org.apache.lucene.search.spans.SpanMultiTermQueryWrapper
         but: <mask(prefix_field._index_prefix:foo) as prefix_field> is a org.apache.lucene.search.spans.FieldMaskingSpanQuery
        at __randomizedtesting.SeedInfo.seed([92425189D86F7872:344C43FEE62E883A]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery(SpanMultiTermQueryBuilderTests.java:206)
```

--- a/server/src/test/java/org/elasticsearch/index/query/SpanMultiTermQueryBuilderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/query/SpanMultiTermQueryBuilderTests.java
@@ -39,7 +39,6 @@ import org.apache.lucene.search.spans.SpanMultiTermQueryWrapper;
 import org.apache.lucene.search.spans.SpanQuery;
 import org.apache.lucene.search.spans.SpanTermQuery;
 import org.apache.lucene.store.Directory;
-import org.elasticsearch.Version;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.compress.CompressedXContent;
 import org.elasticsearch.common.io.stream.StreamOutput;
@@ -53,9 +52,9 @@ import java.io.IOException;
 import static java.util.Collections.singleton;
 import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
 import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.either;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.CoreMatchers.instanceOf;
-import static org.hamcrest.CoreMatchers.either;
 import static org.hamcrest.CoreMatchers.startsWith;
 
 public class SpanMultiTermQueryBuilderTests extends AbstractQueryTestCase<SpanMultiTermQueryBuilder> {
@@ -189,28 +188,18 @@ public class SpanMultiTermQueryBuilderTests extends AbstractQueryTestCase<SpanMu
         assertThat(query, instanceOf(SpanQuery.class));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/58199")
     public void testToQueryInnerTermQuery() throws IOException {
         String fieldName = randomFrom("prefix_field", "prefix_field_alias");
         final QueryShardContext context = createShardContext();
         {
             Query query = new SpanMultiTermQueryBuilder(new PrefixQueryBuilder(fieldName, "foo")).toQuery(context);
-            if (context.getIndexSettings().getIndexVersionCreated().onOrAfter(Version.V_6_4_0)) {
-                assertThat(query, instanceOf(FieldMaskingSpanQuery.class));
-                FieldMaskingSpanQuery fieldQuery = (FieldMaskingSpanQuery) query;
-                assertThat(fieldQuery.getMaskedQuery(), instanceOf(SpanTermQuery.class));
-                assertThat(fieldQuery.getField(), equalTo("prefix_field"));
-                SpanTermQuery termQuery = (SpanTermQuery) fieldQuery.getMaskedQuery();
-                assertThat(termQuery.getTerm().field(), equalTo("prefix_field._index_prefix"));
-                assertThat(termQuery.getTerm().text(), equalTo("foo"));
-            } else {
-                assertThat(query, instanceOf(SpanMultiTermQueryWrapper.class));
-                SpanMultiTermQueryWrapper wrapper = (SpanMultiTermQueryWrapper) query;
-                assertThat(wrapper.getWrappedQuery(), instanceOf(PrefixQuery.class));
-                PrefixQuery prefixQuery = (PrefixQuery) wrapper.getWrappedQuery();
-                assertThat(prefixQuery.getField(), equalTo("prefix_field"));
-                assertThat(prefixQuery.getPrefix().text(), equalTo("foo"));
-            }
+            assertThat(query, instanceOf(FieldMaskingSpanQuery.class));
+            FieldMaskingSpanQuery fieldQuery = (FieldMaskingSpanQuery) query;
+            assertThat(fieldQuery.getMaskedQuery(), instanceOf(SpanTermQuery.class));
+            assertThat(fieldQuery.getField(), equalTo("prefix_field"));
+            SpanTermQuery termQuery = (SpanTermQuery) fieldQuery.getMaskedQuery();
+            assertThat(termQuery.getTerm().field(), equalTo("prefix_field._index_prefix"));
+            assertThat(termQuery.getTerm().text(), equalTo("foo"));
         }
 
         {

